lib/map_walker.go

```diff
@@ -0,0 +1,162 @@
+package lib
+
+import (
+	"fmt"
+	"reflect"
+
+	"github.com/mitchellh/copystructure"
+	"github.com/mitchellh/mapstructure"
+	"github.com/mitchellh/reflectwalk"
+)
+
+func MapWalk(input interface{}) (map[string]interface{}, error) {
```

Comment:
Would be nice to have a doc comment for a lib to make it clear what it's good for etc. The comment below seems to be implementation detail more than usage guidance so maybe a simple doc here would be useful? Maybe include a warning that it does a deep copy of the object as a performance tip.

Reply:
Will do